api/stream.go

```diff
@@ -894,3 +896,45 @@ func (r streamRoutes) updateChatEnabled(c *gin.Context) {
 		return
 	}
 }
+
+func (r streamRoutes) putCustomLiveThumbnail(c *gin.Context) {
+	tumLiveContext := c.MustGet("TUMLiveContext").(tools.TUMLiveContext)
+	streamID := tumLiveContext.Stream.ID
+	course := tumLiveContext.Course
+	file, err := c.FormFile("file")
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid file"})
+		return
+	}
+
+	filename := file.Filename
+	fileUuid := uuid.NewV1()
+
+	filesFolder := fmt.Sprintf("%s/%s.%d/%s.%s/files",
+		tools.Cfg.Paths.Mass,
+		course.Name, course.Year,
+		course.Name, course.TeachingTerm)
+
+	path := fmt.Sprintf("%s/%s%s", filesFolder, fileUuid, filepath.Ext(filename))
```

Comment:
please use path.Join from the standard library to construct this filepath:
https://pkg.go.dev/path#example-Join